projects/storefrontlib/src/cms-components/navigation/navigation/navigation.component.service.ts

```diff
@@ -14,16 +13,55 @@ export class NavigationComponentService {
     protected componentData: CmsComponentData<CmsNavigationComponent>
   ) {}
 
+  public getComponentData(): Observable<CmsNavigationComponent> {
+    return this.componentData.data$;
+  }
+
+  public createNavigation(): Observable<NavigationNode> {
+    return combineLatest(
+      this.getComponentData(),
+      this.getNavigationNode()
+    ).pipe(
+      map(([data, nav]) => {
+        return {
+          title: data.name,
+          children: [nav],
+        };
+      })
+    );
+  }
+
+  public getNavigationNode(): Observable<NavigationNode> {
+    return this.getComponentData().pipe(
+      switchMap(data => {
+        if (data) {
```

Comment:
you should always switchMap to some stream to avoid potential errors. In this case you will only switch when you have the `data`, so either we don't need that 'if', or we should return some stream (or empty) if there is no data